### PR TITLE
Search filter

### LINKED ISSUE #331: [SEARCH] Implement search filter
## Description

Add a search filter on the sidebar of the search results.

## Screenshots
filter

side

## Files

[Zeplin Designs](https://zpl.io/brEmod7)

## Tasks
The highlighted task should follow the design layout provide on zeplin :arrow_up_small: 
- [x] Closable popover
- [x] Create closable tags 
- [x] Filter by country 
- [x] Create filter by keyword input box
- [x] Filter by SDG indicators
- [x] Filter by topics
- [x] Popover for multiple  select items

### LINKED ISSUE #335: [FEATURE] Add pagination on search results page
## Description

Search results pagination for easy navigation between the results page.  

## Screenshots
pagenate

## Files
[Zeplin Design](https://app.zeplin.io/project/5b1a4192c1a8233544fcc66a/screen/5b4405a75792d5de5f566922)

## Tasks

- [x] Add pagination links
- [x] Previous and next button
- [x] Result per page select box
- [x] Style according to zeplin design

## Related Issues
#334 #333 #332 #331 

## Additional Notes
This issue depends on #333

### PR DESCRIPTION
**Pull request by mtuchi**
## Description

Add search results filtering capability on the frontend.

Fixes #335 #331 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
filter

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: takwimu/templates/takwimu/_layouts/_base.html [hunk 1]
```diff
@@ -77,7 +77,27 @@
...
+          $popup.find('#reset-filter').click(function(e){
+            var _parent = $(this).attr("data-parent");
+            $("#results-table tr#result").show();
+            $("input[data-parent="+_parent+"]:checkbox:checked").prop("checked", false);
+            
+            
+            $("#"+_parent+" button .select-item").removeClass("d-none").addClass("d-flex");            
+            $("#"+_parent+" button .filter-badge").removeClass("d-flex").addClass("d-none");
+          });
+      });
```
Comment (DavidLemayian) [inline takwimu/templates/takwimu/_layouts/_base.html]: Won't this trigger for all popover implementations on the site? Selector should be limited to the element being manipulated in this custom implementation.

Comment (mtuchi) [reply takwimu/templates/takwimu/_layouts/_base.html]: @DavidLemayian 
I have pushed an update where i separated the reset filter action to it's component, this way it won't be triggered in other popover implementation

File: takwimu/templates/takwimu/_includes/search/_popover.html [hunk 2]
```diff
@@ -16,4 +32,70 @@
...
+    // Reset filters on popover
+    $(".popover").find("#reset-filter").click(function(e){
+        var _parent = $(this).attr("data-parent");
+        $("#results-table tr#result").show();
+        $("input[data-parent="+_parent+"]:checkbox:checked").prop("checked", false);
+
+
+        $("#"+_parent+" button .select-item").removeClass("d-none").addClass("d-flex");            
+        $("#"+_parent+" button .filter-badge").removeClass("d-flex").addClass("d-none");
+    });
```
Comment (DavidLemayian) [inline takwimu/templates/takwimu/_includes/search/_popover.html]: Every element with the popover class will be affected by this. This class included - https://github.com/TakwimuAfrica/TAKWIMU/pull/374/files#diff-a50a162bb144b49aaf29732276df3ad1R23

Please change.

File: takwimu/templates/takwimu/_includes/search/index.html [hunk 3]
```diff
@@ -63,72 +68,80 @@ <h1 class="display-4 text-center font-weight-bold">Please type something into th
...
+        white-space: -pre-wrap;      /* Opera 4-6 */
+        white-space: -o-pre-wrap;    /* Opera 7 */
+        white-space: pre-wrap;       /* css-3 */
+        word-wrap: break-word;       /* Internet Explorer 5.5+ */
+        white-space: -webkit-pre-wrap; /* Newer versions of Chrome/Safari*/
+        word-break: break-all;
+        white-space: normal;
+    }
+
+</style>
```
Comment (DavidLemayian) [inline takwimu/templates/takwimu/_includes/search/index.html]: This looks like should be globally available and therefore would be better suited in `takwimu.scss` if not already.

File: takwimu/templates/takwimu/_includes/search/pagination.html [hunk 4]
```diff
@@ -15,13 +15,13 @@
...
                         <a class="page-link mx-1">
-                            Prev 
+                            Previous 
                             <span class="sr-only">(current)</span>
                         </a>
                     </li>
                     <!--Here the JS Function Will Add the Rows -->
-                    <li class="page-item" data-page="next" id="prev">
-                        <a class="page-link mx-1 active"> Next
+                    <li class="page-item active" data-page="next" id="prev">
```
Comment (DavidLemayian) [inline takwimu/templates/takwimu/_includes/search/pagination.html]: What if it's only one page of results?

File: takwimu/templates/takwimu/_includes/search/_popover.html (hunk 2 above)
Comment (mtuchi) [reply takwimu/templates/takwimu/_includes/search/_popover.html]: @DavidLemayian 
Only the popover that extends this component will be affected, not in a global level. but only the elements extending this template

File: takwimu/templates/takwimu/_includes/search/pagination.html (hunk 4 above)
Comment (mtuchi) [reply takwimu/templates/takwimu/_includes/search/pagination.html]: @DavidLemayian 
It wont show the pagination, the pagination script handle that

File: takwimu/templates/takwimu/_includes/search/index.html (hunk 3 above)
Comment (mtuchi) [reply takwimu/templates/takwimu/_includes/search/index.html]: @DavidLemayian 
Moved the css for checkbox in` takwimu.scss`

**Pull request merged**